### ISSUE TITLE
java.util.logging.Handler.close() not called on shutdown 

### ISSUE THREAD

**Issue by dansiviter** [oracle/graal#3218]
**Describe the issue**
The shutdown hook on the `java.util.LogManager` does not appear to be run even though `--install-exit-handlers` is set when building. I checked this by creating a custom `java.util.logging.Handler` which logs to console. 

In a containerised environment I noticed some messages can be lost as they are not being flushed on shutdown making debugging problematic. It's also important for implementations that asynchronously flush log messages such as to Cloud Logging/Stackdriver or similar aggregator.

**Steps to reproduce the issue**
Please include both build steps as well as run steps
1. `git clone --depth 1 https://github.com/dansiviter/graal-logger.git`
2. `docker build -t graal-logger .`
3. `docker run --rm -it graal-logger`
4. Observe STDOUT output

**Describe GraalVM and your environment:**
 - GraalVM version: CE 21.0.0
 - JDK major version: 11
 - OS: Docker (via Docker for Desktop Windows)
 - Architecture: Docker x64

**More details**
Successful running `foo.Main`:
```
<init> class foo.MyHandler
Feb 16, 2021 8:33:59 PM foo.Main main
INFO: Hello!
Closing class foo.MyHandler
```

Failure via native-image:
```
<init> class foo.MyHandler
Feb 16, 2021 8:35:21 PM foo.Main main
INFO: Hello!
```

**Comment by mcraj017** [comment]
@dansiviter Thanks for reporting the issue 
We will check it out and get back to you

**Comment by mcraj017** [comment]
@dansiviter I have raised bug to dev team, it will get fixed in future releases

**Comment by teshull** [comment]
Note that `--install-exit-handlers` is meant to install Signal Handlers (`jdk.internal.misc.Signal`) not `java.util.logging.Handler`

**Comment by dansiviter** [comment]
@teshull If you look at [`LogManager#301`](/AdoptOpenJDK/openjdk-jdk11/blob/master/src/java.logging/share/classes/java/util/logging/LogManager.java#L301) it installs a shutdown hook which calls `java.util.logging.Handler#close` and this appears to not get called in older version. However, trying with 21.1.0 it appears it does get called so this may have been fixed.

**Comment by teshull** [comment]
Thanks for the pointer. I believe this issue is still present. The problem is that the shutdown hooks are reset at run time, cleaning the ApplicationShutdownHook created via the LogManager (which is initialized at build time)

A dirty workaround would be to install your own application shutdown hook which calls `LogManager.getLogManager().reset()`, but I'm trying to come up with a better solution presently

**Comment by teshull** [comment]
As a commit 805e93110c92577677f1b7f9a173ecc1f8dfe452 a shutdown hook will be automatically registered for the LogManager the first time `getLogManager` is called. This fix will be part of the 21.3 release
